instructors.md

```diff
@@ -90,18 +90,548 @@ which are covered in the lesson on the shell.
 
+Solutions to exercises:
+> ## Combining strings {.challenge}
+>
+> Write a function called `fence` that takes two parameters called `original` and `wrapper`
+> and returns a new string that has the wrapper character at the beginning and end of the original.
+>
+> ~~~ {.output}
+> def fence(original, wrapper):
+>     return wrapper + original + wrapper
+> ~~~
+
+> ## Selecting characters from strings {.challenge}
+>
+> Write a function called `outer`
+> that returns a string made up of just the first and last characters of its input.
+>
+> ~~~ {.output}
+> def outer(input_string):
+>     return input_string[0] + input_string[-1]
+> ~~~
+
+> ## Rescaling an array {.challenge}
+>
+> Write a function `rescale` that takes an array as input
+> and returns a corresponding array of values scaled to lie in the range 0.0 to 1.0.
+> (Hint: If $L$ and $H$ are the lowest and highest values in the original array,
+> then the replacement for a value $v$ should be $(v-L) / (H-L)$.)
+>
+> ~~~ {.output}
+> def rescale(input_array):
+>     L = input_array.min()
+>     H = input_array.max()
+>     output_array = (input_array - L)/float(H-L)
+>     return output_array
+> ~~~
+
+> ## Testing and documenting your function {.challenge}
+>
+> Run the commands `help(numpy.arange)` and `help(numpy.linspace)`
+> to see how to use these functions to generate regularly-spaced values,
+> then use those values to test your `rescale` function.
+> Once you've successfully tested your function,
+> add a docstring that explains what it does. 
+> ~~~ {.output}
+> Examples:
+> rescale(np.arange(0,10.0))
+> array([ 0.        ,  0.11111111,  0.22222222,  0.33333333,  0.44444444,
+>        0.55555556,  0.66666667,  0.77777778,  0.88888889,  1.        ])
+> rescale(np.linspace(0,100,5))
+> array([ 0.  ,  0.25,  0.5 ,  0.75,  1.  ])
+> '''sample docstring:
+>    takes an array as input, and returns a corresponding array scaled so
+>    that 0 corresponds to the minimum and 1 to the maximum value of the input array'''
+> ~~~
+
+
+> ## Defining defaults {.challenge}
+>
+> Rewrite the `rescale` function so that it scales data to lie between 0.0 and 1.0 by default,
+> but will allow the caller to specify lower and upper bounds if they want.
+> Compare your implementation to your neighbor's:
+> do the two functions always behave the same way?
+> ~~~ {.output}
+> def rescale(input_array, low_val = 0.0, high_val = 1.0):
+>     '''rescales input array values to lie between low_val and high_val'''
+>     L = input_array.min()
+>     H = input_array.max()
+>     intermed_array = (input_array - L)/float(H-L) 
+>     output_array = intermed_array*(high_val-low_val) + low_val
+>     return output_array
+> ~~~
+
+> ## Variables inside and outside functions {.challenge}
+>
+> What does the following piece of code display when run - and why?
+>
+> ~~~ {.python}
+> f = 0
+> k = 0
+>
+> def f2k(f):
+>   k = ((f-32)*(5.0/9.0)) + 273.15
+>   return k
+>
+> f2k(8)
+> f2k(41)
+> f2k(32)
```

Comment:
We probably want print statements in front of the f2k for those people doing this in a single cell in the notebook